### PR TITLE
Fix issues with test_new_model and date_release ci jobs

### PR DESCRIPTION
**Pull request by haimasree**
job test_new_model was not working due to something going wrong with the way circleci saves and restores cache. I added a new test model and checked if  ```kipoi test-source kipoi --git-range master HEAD --verbose``` is identifying the newly added model correctly. Check [here](https://app.circleci.com/pipelines/github/kipoi/models/453/workflows/be23426d-d871-419e-8a18-0859145cd6d5/jobs/1870)

I also wanted to replicate the problem of Xpresso model not triggering a build [here](https://github.com/kipoi/models/pull/228) and [here](https://app.circleci.com/pipelines/github/kipoi/models/456/workflows/c348e1c4-0a82-41f5-8da9-b5752a1abba3/jobs/1872). As you can see, I could not replicate the problem. My hypothesis is: test_new_model did not get triggered which led to a sort of a limbo state in circleci server but admittedly this does not explain why I was unable to find the job. 

Also, I fixed issue with failing date_release job.

Before I push this PR to master, maybe you can tell me if there was any purpose other than speedup behind using cirecleci caching mechanism? I spent a good deal of time in trying to pinpoint what exactly is going wrong but was unable to do so. So, if speedup was the only reason may I suggest removing it? @Avsecz

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,7 +51,6 @@ jobs:
       - checkout
       - *install_sys_deps
       - *install_gitlfs
-      - *restore_cache
```
Comment (Avsecz) [inline .circleci/config.yml:67]: I'd keep the cache when testing all the models.

File: .circleci/config.yml [hunk 2]
```diff
@@ -10,23 +10,12 @@ variables:
   install_sys_deps: &install_sys_deps
     run:
       name: install build-essential
-      command: apt-get update && apt-get install -y build-essential libz-dev libcurl3-dev
+      command: apt-get update && apt-get install -y build-essential libz-dev libcurl3-dev curl
   install_gitlfs: &install_gitlfs
     run:
       name: Install git-lfs
       command: conda install --yes -c conda-forge git-lfs && git lfs install
-  restore_cache: &restore_cache
```
Comment (Avsecz) [inline .circleci/config.yml:18]: I'd keep the caching functionality when testing for all the models.

**Comment by Avsecz** [comment]
I think the reason was that it took a while to download/checkout all the models. I think this was from the times when we were still using git-lfs to download the model weights. Can you comment-out the caching functionally for test_all_models? We can see the speed decrease of test_all_models, and if if it's much slower add it back.

File: .circleci/config.yml (hunk 1 above)
Comment (haimasree) [reply .circleci/config.yml:67]: Yes ofcourse
Here - https://github.com/kipoi/models/commit/aaf51c17eabb07f703dc2f29a9a507550301e0b4

File: .circleci/config.yml (hunk 2 above)
Comment (haimasree) [reply .circleci/config.yml]: Yes done.

**Pull request merged**